src/providers/mod.rs

```diff
@@ -152,6 +154,16 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
     Ok(())
 }
 
+fn max_hostname_len() -> Result<Option<usize>> {
+    unistd::sysconf(unistd::SysconfVar::HOST_NAME_MAX)
```

Comment:
This is all fine as is, but at least systemd does it via compile-time and not runtime constant.  OTOH, I guess querying dynamically is perhaps cleaner.

Reply:
Yeah, I'd prefer to use a compile-time constant.  Any idea where I can get one?

Reply:
I think for libc it's done manually, see e.g. https://github.com/rust-lang/libc/pull/2580
which is a random one I found.  I guess no one else needed it yet.

Reply:
I did some digging.  At least FreeBSD and Mac OS X don't define `HOST_NAME_MAX`, which is allowed by POSIX if the value is at least 255.  So I think there's not much point in defining it in `libc`, since portable code couldn't assume the constant is available.

systemd doesn't have to be portable, of course.  We could hardcode a value here (or non-portably in `libc`), but it's ambiguous whether Afterburn supports non-Linux, so... sticking with the current approach for now.